src/tilemap.rs

```diff
@@ -1,13 +1,18 @@
 use std::collections::BTreeMap;
 
-use bevy::prelude::*;
+use bevy::{gltf::Gltf, prelude::*};
+
+enum LoadSource {
+    Local,
+    Network,
+}
 
 #[derive(Component, Default)]
 pub struct TileMap<const TILE_SIZE: u32> {
     /// All currently loaded tiles.
     tiles: BTreeMap<i32, BTreeMap<i32, Entity>>,
     /// The tile currently being loaded.
-    loading: Option<(IVec2, Handle<Scene>)>,
+    loading: Option<(IVec2, Handle<Gltf>, LoadSource)>,
```

Comment:
This change from Scene to Gltf is necessary because I was unable to implement gz decompression for arbitrary handles. We'll be able to switch back to `Scene` in the future